ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,898 @@
+            drillNumb = defined(drillAmount) ?
+                breadcrumbsList[breadcrumbsList.length - 1][0] - drillAmount :
+                breadcrumbsList[breadcrumbsList.length - 1][0] + 1;
+
+        if (!this.options.showOnlyLast) {
+            if (breadcrumbsList && breadcrumbsList.length) {
+                for (let i = 0; i < drillNumb; i++) {
+                    if (this.isTreemap) {
+                        (chart.series[0] as Highcharts.TreemapSeries).drillUp();
+                    } else {
+                        chart.drillUp();
+                    }
+                }
+            }
+        } else {
+            if (this.isTreemap) {
+                (chart.series[0] as Highcharts.TreemapSeries).drillUp();
+            } else {
+                chart.drillUp();
+            }
+        }
+
+        this.destroy(drillAmount);
+    }
+
+    /**
+    * Redraw.
+    *
+    * @requires  modules/breadcrums
+    *
+    * @function Highcharts.Breadcrumbs#redraw
+    * @param {Highcharts.Breadcrumbs} this
+    *        Breadcrumbs class.
+    */
+    public redraw(this: Breadcrumbs): void {
+        const breadcrumbsGroup = this.breadcrumbsGroup;
+
+        if (breadcrumbsGroup && this.breadcrumbsList.length) {
+            this.alignGroup();
+        }
+        this.createList();
+        this.draw();
+    }
+
+    /**
+    * Render the button.
+    *
+    * @requires  modules/breadcrums
+    *
+    * @function Highcharts.Breadcrumbs#renderButton
+    * @param {Highcharts.Breadcrumbs} this
+    *        Breadcrumbs class.
+    * @param {Highcharts.Breadcrumbs} breadcrumb
+    *        Current breadcrumb
+    * @param {Highcharts.Breadcrumbs} posX
+    *        Initial horizontal position
+    * @param {Highcharts.Breadcrumbs} posY
+    *        Initial vertical position
+    * @return {SVGElement|void}
+    *        Returns the SVG button
+    */
+    public renderButton(this: Breadcrumbs, breadcrumb: Array<any>, posX: number, posY: number): SVGElement|undefined {
+        const breadcrumbs = this,
+            chart = this.chart,
+            breadcrumbsList = breadcrumbs.breadcrumbsList,
+            breadcrumbsOptions = chart.options.breadcrumbs,
+            lang = chart.options.lang;
+        let arrow = '';
+
+        if (breadcrumbsOptions && breadcrumb && breadcrumb[1]) {
+            if (breadcrumbsOptions && breadcrumbsOptions.showOnlyLast) {
+                arrow = '◁ ';
+            }
+
+            const button = chart.renderer.button(
+                breadcrumbsOptions.formatter ? breadcrumbsOptions.formatter(breadcrumb, breadcrumbs) : void 0 ||
+                    format(arrow + breadcrumbsOptions.format as string,
+                        { value: breadcrumb[1].name || (lang && lang.mainBreadCrumb) }, chart),
+                posX,
+                posY,
+                function (e: (Event|Highcharts.Dictionary<any>)): void {
+                    // extract events from button object and call
```

Comment:
In general, when we write a full-line comment, we start it with a capital letter, for example:

```js
// My comment
doSomething();
```
But this should not use capital letter:
```js
doSomething(); // my comment
```